analysis/src/CompletionFrontEnd.ml

```diff
@@ -40,8 +72,16 @@ type jsxProps = {
   childrenStart: (int * int) option;
 }
 
+(* Checks if an expression is elgible for typed context completion. *)
+let isElgibleForTypedContextCompletion exp =
```

Comment:
Maybe this is just an artifact of the current approach being semi-wrong in some way, but I feel I need to have something like this that prevents me from "eating" expressions for type based completion, that shouldn't get completed. 

This essentially means that type based completions only work on one level and not deeper into the tree. I have a few ideas on how we can make this more robust and work in a more generalized way. But, as I said in the PR description, I'm letting that rest for just a little while longer.

Reply:
I think it's true that you only want to proceed if an interesting type is found. Where interesting means: actionable for completion. If not, it's as if noting was seen.

That said, sub-expressions override outer expressions, so you don't need to hold back early on if you know something more precise will be found.

Reply:
I don't want to actually implement it now, but I want the final solution to make this kind of completion simple too:

```rescript
type someVariant = One | Two | Three

let someFunc = (~someParam: someVariant) => ...

someFunc(~someParam={
  if whatever == 123 {
     
//. ^com
})
```
If possible, I want the above to complete to `someVariant`, and I hope the rule could be as simple as "if this is the last expression in the braces body, check what type this block should return. Oh it's someVariant, complete from that".